exp/ipam/api/v1beta1/condition_consts.go

```diff
@@ -0,0 +1,13 @@
+package v1beta1
+
+// These reasons are intended to be used with failed Ready conditions on an [IPAddressClaim].
+const (
+	// AllocationFailedReason is the reason used when allocating an IP address for a claim fails. More details should be
+	// provided in the condition's message.
+	// When the IP pool is full, [PoolExhaustedReason] should be used for better visibility instead.
+	AllocationFailedReason = "AllocationFailed"
+
+	// PoolExhaustedReason is the reason used when an IP pool referenced by an [IPAddressClaim] is full and no address
+	// can be allocated for the claim.
+	PoolExhaustedReason = "PoolExhausted"
+)
```

Comment:
I was also thinking about a `PoolNotReady` reason to make debugging more straight forward when using ready conditions on pools (our [infoblox provider (still wip but public now)](https://github.com/telekom/cluster-api-ipam-provider-infoblox) does for example).

Reply:
To clarify, to have a `PoolNotReady` reason that would be set in conditions of IPPools, or on IPAddressClaims?